libs/ngx-dynamic-form-builder/src/lib/utils/dynamic-form-group.ts

```diff
@@ -406,115 +143,498 @@ export class DynamicFormGroup<TModel> extends FormGroup {
       throw error;
     }
   }
+
   setValidatorOptions(validatorOptions: ValidatorOptions) {
     this.setValidatorOptionsAsync(validatorOptions).then(() => { }, error => { throw error; });
   }
+
   getValidatorOptions(): ValidatorOptions {
     return this._validatorOptions;
   }
-  private mergeErrors(errors?: IShortValidationErrors, externalErrors?: IShortValidationErrors) {
+
+  // Helpers
+  private onlyFields(fields: Dictionary) {
+    const newFields = {};
+
+    if (fields !== undefined) {
+      Object.keys(fields).forEach(key => {
+        if (fields[key] instanceof DynamicFormGroup) {
+          // Group: recursive
+          newFields[key] = this.onlyFields((fields[key] as DynamicFormGroup<any>).fields);
+        }
+        else {
+          // Array
+          if (fields[key] instanceof FormArray) {
+            if ((fields[key] as FormArray).controls[0] instanceof DynamicFormGroup) {
+              // Group within Array: recursive
+              newFields[key] = this.onlyFields(((fields[key] as FormArray).controls[0] as DynamicFormGroup<any>).fields);
+            }
+            else {
+              // Control within Array
+              newFields[key] = (fields[key] as FormArray).controls[0].value;
+            }
+          }
+          else {
+            // Handle Control
+            newFields[key] = fields[key];
+          }
+        }
+      });
+    }
+
+    return newFields;
+  }
+
+  private transformValidationErrors(errors: ValidationError[]): ShortValidationErrors {
+    const customErrors: ShortValidationErrors = {};
+
+    errors.forEach((error: ValidationError) => {
+      if (error && error.constraints !== undefined) {
+        Object.keys(error.constraints).forEach((key: string) => {
+
+          if (!customErrors[error.property]) {
+            customErrors[error.property] = [];
+          }
+
+          if ((customErrors[error.property] as string[]).indexOf(error.constraints[key]) === -1) {
+            (customErrors[error.property] as string[]).push(error.constraints[key]);
+          }
+        });
+      }
+
+      if (error.children !== undefined && error.children.length) {
+        customErrors[error.property] = this.transformValidationErrors(error.children);
+      }
+    });
+
+    return customErrors;
+  }
+
+  private mergeErrors(errors?: ShortValidationErrors, externalErrors?: ShortValidationErrors) {
     return mergeWith(errors, externalErrors, (objValue, srcValue) => {
-      if (
-        Array.isArray(objValue) &&
-        Array.isArray(srcValue) &&
-        objValue.filter(objItem => srcValue.indexOf(objItem) !== -1).length === 0
-      ) {
+      if (canMerge()) {
```

Comment:
Moved `if()` statement to a named local function to describe what it is checking